controllers/grid/users/userSelect/UserSelectGridHandler.inc.php

```diff
@@ -0,0 +1,208 @@
+				array('width' => 5)
+			)
+		);
+		$this->addColumn(
+			new GridColumn(
+				'name',
+				'author.users.contributor.name',
+				null,
+				null,
+				$cellProvider,
+				array('alignment' => COLUMN_ALIGNMENT_LEFT,
+					'width' => 30
+				)
+			)
+		);
+	}
+
+
+	//
+	// Overridden methods from GridHandler
+	//
+	/**
+	 * @copydoc GridHandler::initFeatures()
+	 */
+	function initFeatures($request, $args) {
+		import('lib.pkp.classes.controllers.grid.feature.InfiniteScrollingFeature');
+		import('lib.pkp.classes.controllers.grid.feature.CollapsibleGridFeature');
+		return array(new InfiniteScrollingFeature('infiniteScrolling', $this->getItemsNumber()), new CollapsibleGridFeature());
+	}
+
+	/**
+	 * @copydoc GridHandler::loadData()
+	 */
+	protected function loadData($request, $filter) {
+		list($filterUserGroupId, $name) = $this->getFilterValues($filter);
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$stageId = $this->getAuthorizedContextObject(ASSOC_TYPE_WORKFLOW_STAGE);
+		$rangeInfo = $this->getGridRangeInfo($request, $this->getId());
+		$userStageAssignmentDao = DAORegistry::getDAO('UserStageAssignmentDAO'); /* @var $userStageAssignmentDao UserStageAssignmentDAO */
+		return $userStageAssignmentDao->filterUsersNotAssignedToStageInUserGroup($submission->getId(), $stageId, $filterUserGroupId, $name, $rangeInfo);
+	}
+
+	/**
+	 * @copydoc GridHandler::renderFilter()
+	 */
+	function renderFilter($request, $filterData = array()) {
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$stageId = $this->getAuthorizedContextObject(ASSOC_TYPE_WORKFLOW_STAGE);
+
+		$allFilterData = array_merge(
+			$filterData,
+			array(
+				'userGroupOptions' => $this->_userGroupOptions,
+				'selectedUserGroupId' => reset(array_keys($this->_userGroupOptions)),
+				'gridId' => $this->getId(),
+				'submissionId' => $submission->getId(),
+				'stageId' => $stageId,
+			));
+		return parent::renderFilter($request, $allFilterData);
+	}
+
+	/**
+	 * @copydoc GridHandler::getFilterSelectionData()
+	 */
+	function getFilterSelectionData($request) {
+		$name = (string) $request->getUserVar('name');
+		$filterUserGroupId = (int) $request->getUserVar('filterUserGroupId');
+		return array(
+			'name' => $name,
+			'filterUserGroupId' => $filterUserGroupId,
+		);
+	}
+
+	/**
+	 * @copydoc GridHandler::getFilterForm()
+	 */
+	protected function getFilterForm() {
+		return 'controllers/grid/users/userSelect/searchUserFilter.tpl';
+	}
+
+	/**
+	 * @copydoc GridHandler::getRequestArgs()
+	 */
+	function getRequestArgs() {
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$stageId = $this->getAuthorizedContextObject(ASSOC_TYPE_WORKFLOW_STAGE);
+		return array(
+			'submissionId' => $submission->getId(),
+			'stageId' => $stageId,
+		);
+	}
+
+	/**
+	 * Define how many items this grid will start loading.
+	 * @return int
+	 */
+	protected function getItemsNumber() {
+		return 20;
+	}
+
```

Comment:
No problem, I also thought so, but then I also thought, you will surely tell me if needed... :)